### PR TITLE
Turbine-js feature branch changes (--listresources and app name)

### PR DESCRIPTION
**Pull request by anna-cross**
# Description

Changing list resources to output a json with resource information (now containing collection, and if a resource is source or destination) 

EX: 

`[
  {
    "name": "source", 
    "source": true, 
    "destination": false, 
    "collection": "orders"
  }, 
  {
    "name": "dest", 
    "source": false, 
    "destination": true, 
    "collection": "test"
  }
]`

Additionally, the new change also requires an app_name passed to clideploy, so we can override app.json value. 

**Before - regularly running jsapp:**

image

**After - overriding app.json for same app:** 

image

As part of  (https://github.com/meroxa/product/issues/398)

## Type of change

- [x] Refactor

# How Has This Been Tested?

1. Grab changes 
2. In the turbine-js repo, run : tsc -p tsconfig.json
3. Then : npm install turbine-js/ -U 
4. Grab meroxa-js changes 
5. In the meroxa-js repo, run: tsc -p tsconfig.json
6. Then: npm install meroxa-js/ -U 
7. Log into staging meroxa on cli 
8. Get your token 
9. meroxa config --json | jq -r .config.access_token
10. SET - export MEROXA_ACCESS_TOKEN="tokebabove" 
11. SET - export MEROXA_API_URL="https://api.staging.meroxa.io/v1"
12. Now you can run tests! 
13. Create a jsapp or use existing one
14. Git your gitsha in app's dir by running -  git rev-parse HEAD
15. Try: node turbine-js/bin/turbine-js clideploy "meroxa/funtime-example-noop" YOURAPPDIR/  APPNAME GITSHA
16. Try: node turbine-js/bin/turbine-js listresources YOURAPPDIR/ 

- [x] Manual Tests

### CODE REVIEW

File: src/runner/base.ts [hunk 1]
```diff
@@ -9,12 +9,14 @@ import { InfoRuntime } from "../runtime/info";
...
   meroxaJS: MeroxaJS;
+  appName: string;
   infoRuntime: InfoRuntime;
   localRuntime: LocalRuntime;
 
-  constructor(pathToDataApp: string, meroxaJS: MeroxaJS) {
+  constructor(pathToDataApp: string, meroxaJS: MeroxaJS, name: string="") {
     this.pathToDataApp = pathToDataApp;
     this.meroxaJS = meroxaJS;
+    this.appName = name
```
Comment (jmar910) [inline src/runner/base.ts:19]: nitpick: `;` at the end here

File: src/runner/base.ts [hunk 2]
```diff
@@ -26,7 +28,9 @@ export default class Base {
   }
 
   get appJSON() {
-    return require(path.resolve(`${this.pathToDataApp}/app.json`));
+    var config = require(path.resolve(`${this.pathToDataApp}/app.json`))
+    config.name = this.appName
```
Comment (jmar910) [inline src/runner/base.ts:32]: ditto `;`

Comment (anna-cross) [reply src/runner/base.ts]: Updated!! :D

File: src/runtime/info.ts [hunk 3]
```diff
@@ -1,8 +1,9 @@
-import { AppConfig, Records, RecordsArray, RegisteredFunctions } from "./types";
+import { throws } from "assert";
+import { AppConfig, Records, Resource, RecordsArray, RegisteredFunctions } from "./types";
 
 export class InfoRuntime {
   registeredFunctions: RegisteredFunctions = {};
-  registeredResources: string[] = [];
+  registeredResources: InfoResource[] = [];
```
Comment (jayjayjpg) [inline src/runtime/info.ts:6]: Nice refactor! :sparkles:

**Review by jayjayjpg — APPROVED** [review]
This reads great, awesome work! :sparkles:

**Comment by jmar910** [comment]
@anna-cross run `npm run prettier:makeitpretty` from the project root and it should prettify everything to pass CI

**Comment by anna-cross** [comment]
> @anna-cross run `npm run prettier:makeitpretty` from the project root and commit the changes. it should prettify everything to pass CI

I'll give that a try! Thanks!!

**Review by jmar910 — APPROVED** [review]

**Pull request merged**